### PR TITLE
feat: wiring offer cleanup

### LINKED ISSUE #10898: Find solution for offering reserved states
When passing around states through the wiring framework, a reservation on the state is made for each receiver. Some components use `offer` when passing states. This presents a problem because the reservation must be cancelled if the offer fails.

Example:
```
        final boolean success = asyncLatestCompleteStateQueue.offer(reservedSignedState);
        if (!success) {
            reservedSignedState.close();
        }
```

Find a solution to support this paradigm.

### LINKED ISSUE #10924: Add ability to do cleanup of signed state if offer is rejected
In the wiring framework, if we are passing a signed state to a destination using offer soldering, we need to release the reference count on the state if the destination rejects the state.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #10924
closes #10898

### CODE REVIEW

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/wires/output/internal/TransformingOutputWire.java [hunk 1]
```diff
@@ -43,27 +45,32 @@ public class TransformingOutputWire<IN, OUT> extends ForwardingOutputWire<IN, OU
...
+     * @param model         the wiring model containing this output wire
+     * @param name          the name of the output wire
+     * @param transformer   the function to transform the data from the input type to the output type. Is called once
+     *                      per output per data item. If this method returns null then the data is not forwarded.
+     * @param inputCleanup  an optional method that is called on input data after the data is forwarded to all
+     *                      destinations. The original data is passed to this method. Ignored if null.
+     * @param outputCleanup an optional method that is called on output data if it is rejected by a destination. This is
+     *                      possible if offer soldering is used and the destination declines to take the data.
      */
     public TransformingOutputWire(
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/wires/output/internal/TransformingOutputWire.java:63]: I feel like this method should be removed. We can still support the lambda approach with a helper class. @cody-littley wdyt?

Comment (lpetrovic05) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/wires/output/internal/TransformingOutputWire.java:63]: Agreed in zoom call

Comment (lpetrovic05) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/wires/output/internal/TransformingOutputWire.java:63]: done

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/transformers/internal/AdvancedWireTransformer.java [hunk 2]
```diff
@@ -27,6 +27,8 @@
 import java.util.function.Consumer;
 import java.util.function.Function;
 
+// TODO unit test output cleanup
```
Comment (alittley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/transformers/internal/AdvancedWireTransformer.java:30]: TODO

**Review by alittley — DISMISSED** [review]

**Review by cody-littley — COMMENTED** [review]
LGTM

**Review by anthony-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/transformers/internal/AdvancedWireTransformer.java (hunk 2 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/transformers/internal/AdvancedWireTransformer.java]: good catch, thanks

**Review by lpetrovic05 — APPROVED** [review]

**Pull request merged**